### ISSUE TITLE
Spikekill | Does not work properly with graphs with 2 or more data sources/templates

### ISSUE THREAD

**Issue by zuka1337** [Cacti/cacti#3648]
# Describe the bug
I'm testing the killspike in several graphs and works like a charm in silge's graphs with single data source.
I've teste in 2 versions of **cacti dev 1.1.12** and **cacti prod 1.2.7**

A clear and concise description of what the bug is:

If you try to repeat the same process with multiple data source in one graph it only erase a single vertical/horizontal Line, ignoring the spike AREA or all the history data is erased and bye bye graph.

## To Reproduce

1. Create 2 datasources, with 2 Areas and 2 lines 
Create a graph with those datasources

![spike_graph](https://user-images.githubusercontent.com/57998098/86106723-d1831900-bab8-11ea-88fb-657375ed8de4.png)

2. Click on Spike and change for any value you want to, the best result I get is this:

![imagem](https://user-images.githubusercontent.com/57998098/86107802-3db24c80-baba-11ea-9a90-fba6b5fb4d9b.png)
![imagem](https://user-images.githubusercontent.com/57998098/86107287-89b0c180-bab9-11ea-8a1e-37b2b794e5e6.png)
![imagem](https://user-images.githubusercontent.com/57998098/86108778-84547680-babb-11ea-90a7-c707f4e132a7.png)

If I keep trying, the graph will broke, and the all the data history erase it self.

## Expected behavior

![Working_Spike](https://user-images.githubusercontent.com/57998098/86111570-d0ed8100-babe-11ea-9fe1-be3e0eb93ee3.gif)

## Desktop (please complete the following information)

- OS: Windows 10 

- Browser
Tested in Chrome, firefox, Eadge, Internet Explorer and Chromium

- Version: All updated.

## Additional context

I have 2 scripts to simulate the spike
1 scripts that generate a Average number between 1 to 10 +1, and a commented 10k value to create spike.
1 scripts that generate a Average number between 40 to 50 +1, and a commented 10k value to create spike.

Logs of (Not Working kilspike graph)

```
2020-06-30  10:51:37 - SPIKEKILL CactiUser:admin, File:cacti_spike_34.rrd,  Method:RangeFill, OutStart:1593510036, OutEnd:1593510341, AvgNan:last
--
2020-06-30 10:51:37 - WEBUI WARNING: Removed '13' Spikes from '/var/www/html/cacti/rra/cacti_spike1_35.rrd'  DS[spike1] Graphs[spike1], Method:'3'
2020-06-30 10:51:37 - SPIKEKILL CactiUser:admin,  File:cacti_spike1_35.rrd, Method:RangeFill, OutStart:1593510036,  OutEnd:1593510341, AvgNan:last
2020-06-30 10:51:35 - SPIKEKILL CactiUser:admin,  File:cacti_spike1_34.rrd, Method:GapFill, OutStart:1593510036,  OutEnd:1593510341, AvgNan:last
2020-06-30 10:51:35 - SPIKEKILL CactiUser:admin,  File:cacti_spike1_35.rrd, Method:GapFill, OutStart:1593510036,  OutEnd:1593510341, AvgNan:last
2020-06-30 10:51:29 - SPIKEKILL CactiUser:admin,  File:cacti_spike1_34.rrd, Method:RangeFill, OutStart:1593509890,  OutEnd:1593510389, AvgNan:last
2020-06-30 10:51:29 - WEBUI WARNING: Removed '16' Spikes from '/var/www/html/cacti/rra/cacti_spike1_35.rrd'  DS[spike1] Graphs[spike1], Method:'3'
2020-06-30 10:51:28 - SPIKEKILL CactiUser:admin,  File:cacti_spike1_35.rrd, Method:RangeFill, OutStart:1593509890,  OutEnd:1593510389, AvgNan:last

```

**Comment by netniV** [comment]
This should be addressed once #3636 is completed.  In fact, there are a couple of related issue trackers on SpikeKill.  So, for now, I'm going to do nothing on this thread but tag the other so it can be referenced.  We are still waiting on another of our users to supply the potential fixes.

**Comment by riversdev0** [comment]
Unfortunately, I don't have a clear understanding of the scenario yet.  I can clearly see in the images that something isn't right, but I'm struggling to understand how the problem exists when it's a graph with multiple data sources, while it does not exist when it's a graph with a single data source.  When SpikeKill is run on a graph with multiple underlying RRDs, the process is run on all DSes inside all associated RRDs.  Furthermore, SpikeKill does not make any distinctions between values inside an RRD based upon whether the values are graphed as a LINE or as an AREA.  Therefore, SpikeKill should work identically on a graph of a single data data source as it does on a graph with a single data source.  Can you please add any clarification that might be helpful for me to understand it better?

**Comment by netniV** [comment]
> SpikeKill should work identically on a graph of a single data data source as it does on a graph with a single data source. Can you please add any clarification that might be helpful for me to understand it better?

Maybe you could clarify if you mean single vs multi there coz it wouldn't be different for a single vs single ;-)

tbh, @TheWitness is a better candidate as I have never really perused the spikekill code.

**Comment by riversdev0** [comment]
Thank you, edited previous comment.

**Comment by zuka1337** [comment]
@riversdev0 thanks for the reply, here we go:

Single data template from one data source, What I'm trying to say is that the graph is creating data from a single source only, in this case I've used a gif image to describe, you can see as "Current". That graph template has only one item "Area" to simplify.

On the other hand with "multiple data sources" or "Multiple data templates" I mean a graph that display 2 or more (data sources) can be for example "Memory", "load Average","CPU" etc, all in one graph.

As I show in the 3 images I've created 4 data tamplates intems 2 "inbound" and 2 "outbound", like that I have 2 informations on the same graph. and if one of them spikes.

4 or 5 days of history data gonne (it's ok couse they are only test scripts, but would be hard on a 7 years old graph )
**For simplification, I've remove the 2 Lines items from the graph.**
 I was testing this again early today, and again all the history data erased itself:

![history_data_erased_Spike](https://user-images.githubusercontent.com/57998098/86452629-7c434380-bd14-11ea-9ba4-f4d0ebab786b.gif)

![not_Spike](https://user-images.githubusercontent.com/57998098/86452780-ba406780-bd14-11ea-9c5c-22b3f9f90b46.gif)


My workaround now is to replace spikes manualy for NaN.

PS: Sorry if I make myself unclear.

**Comment by zuka1337** [comment]
Why can't I make the /cli/removespikes.php work ?

**php /var/www/html/cacti/cli/removespikes.php --rrdfile=/root/2.rrd --method=stddev --avgnan=last --debug**
```
DEBUG: Var Kill: Value '9.9334e+3', VarianceDev '5.3764e+0', VarianceLimit '5.9140e+1'
DEBUG: Var Kill: Value '1.0000e+4', VarianceDev '5.3764e+0', VarianceLimit '5.9140e+1'
DEBUG: Var Kill: Value '7.4613e+1', VarianceDev '5.3764e+0', VarianceLimit '5.9140e+1'
DEBUG: Var Kill: Value '3.3378e+3', VarianceDev '5.2078e+0', VarianceLimit '5.7286e+1'
DEBUG: Var Kill: Value '8.3921e+2', VarianceDev '4.9430e+0', VarianceLimit '5.4373e+1'
DEBUG: Var Kill: Value '9.9334e+3', VarianceDev '5.3764e+0', VarianceLimit '5.9140e+1'
DEBUG: Var Kill: Value '1.0000e+4', VarianceDev '5.3764e+0', VarianceLimit '5.9140e+1'
DEBUG: Var Kill: Value '7.4613e+1', VarianceDev '5.3764e+0', VarianceLimit '5.9140e+1'
DEBUG: Var Kill: Value '9.9334e+3', VarianceDev '5.3764e+0', VarianceLimit '5.9140e+1'
DEBUG: Var Kill: Value '1.0000e+4', VarianceDev '5.3764e+0', VarianceLimit '5.9140e+1'
DEBUG: Var Kill: Value '7.4613e+1', VarianceDev '5.3764e+0', VarianceLimit '5.9140e+1'
DEBUG: Var Kill: Value '1.0000e+4', VarianceDev '8.7323e+0', VarianceLimit '9.6055e+1'
DEBUG: Var Kill: Value '1.0000e+4', VarianceDev '9.7973e+0', VarianceLimit '1.0777e+2'
DEBUG: Var Kill: Value '9.9334e+3', VarianceDev '5.3764e+0', VarianceLimit '5.9140e+1'
DEBUG: Var Kill: Value '1.0000e+4', VarianceDev '5.3764e+0', VarianceLimit '5.9140e+1'
DEBUG: Var Kill: Value '7.4613e+1', VarianceDev '5.3764e+0', VarianceLimit '5.9140e+1'
DEBUG: Var Kill: Value '7.4613e+1', VarianceDev '4.7200e+0', VarianceLimit '5.1920e+1'
NOTE: Creating XML file '/var/www/html/cacti/cache/spikekill//2.dump.18560295' from '/root/2.rrd'
NOTE: Searching for Spikes in XML file '/var/www/html/cacti/cache/spikekill//2.dump.18560295'

      Size       DataSource         CF Samples  NonNan        Avg     StdDev   MaxValue   MinValue  MaxStdDev  MinStdDev  StdKilled  VarKilled WindFilled  StdDevAvg     VarAvg
----------  --------------- ---------- ------- ------- ---------- ---------- ---------- ---------- ---------- ---------- ---------- ---------- ---------- ---------- ----------
    5 mins            spike    AVERAGE     600     520      43.88     574.21      10000          1          0          0          0          3          0          0       5.38
   30 mins            spike    AVERAGE     700      87      43.74     126.06    3337.83       2.84          0          0          0          1          0          0       5.21
   2 hours            spike    AVERAGE     775      21      45.11      30.13     839.21       4.42          0          0          0          1          0          0       4.94
    1 days            spike    AVERAGE     797       1       5.41       0.19       5.41       5.41          0       3.49          0          0          0       5.41          0
    5 mins            spike        MIN     600     520      43.88     574.21      10000          1          0          0          0          3          0          0       5.38
   30 mins            spike        MIN     700      87        2.1       0.83          6          1          0          0          0          0          0          0       1.77
   2 hours            spike        MIN     775      21       1.15        0.2       2.01          1          0          0          0          0          0          0          1
    1 days            spike        MIN     797       1          1       0.04          1          1          0       0.65          0          0          0          1          0
    5 mins            spike        MAX     600     520      43.88     574.21      10000          1          0          0          0          3          0          0       5.38
   30 mins            spike        MAX     700      87     123.71     377.66      10000          5          0          0          0          1          0          0       8.73
   2 hours            spike        MAX     775      21     485.61     358.97      10000       8.98          0          0          0          1          0          0        9.8
    1 days            spike        MAX     797       1         10       0.35         10         10          0       6.46          0          0          0         10          0
    5 mins            spike       LAST     600     520      43.88     574.21      10000          1          0          0          0          3          0          0       5.38
   30 mins            spike       LAST     700      87       6.06       3.43      74.61          1          0          0          0          1          0          0       4.72
   2 hours            spike       LAST     775      21       5.05       0.92       9.99          1          0          0          0          0          0          0        3.1
    1 days            spike       LAST     797       1       5.99       0.21       5.99       5.99          0       3.87          0          0          0       5.99          0

NOTE: NO Standard Deviation found in '/root/2.rrd'
NOTE: No Spikes Found.  No remediation performed.
```

```
php /var/www/html/cacti/cli/removespikes.php --rrdfile=/root/2.rrd --method=fill --avgnan=last
NOTE: Creating XML file '/var/www/html/cacti/cache/spikekill/2.dump.1753731706' from '/root/2.rrd'
NOTE: Searching for Spikes in XML file '/var/www/html/cacti/cache/spikekill/2.dump.1753731706'
NOTE: NO Variance Spikes found in '/root/2.rrd'
NOTE: No Spikes Found.  No remediation performed.
```

**Comment by riversdev0** [comment]
Can you please be more specific when you say that the script doesn't work?  Please provide an explanation of how you expected the results to be different than the results you were given.

**Comment by riversdev0** [comment]
@zuka1337 I have added some debugging capability to the Spikekill script in the last couple weeks.  Can you please do the following:
1. Grab a fresh copy of `lib/spikekill.php` from the `1.2.x` branch on GitHub and place it in your Cacti installation, overwriting the copy you have.
2. In the Cacti Settings, set your 'Generic Log Level' to 'Debug'.  Remember what level it was previously set to.
3. Start screen recording as you have previously done.
4. Run Spikekill on your test graph again.
5. Stop screen recording.
6. Set your 'Generic Log Level' back to whatever it was previously.
7. Gather any log entries written to `log/cacti.log` from Spikekill.  These entries will contain the string 'SPIKE'.
8. Upload the screen recording GIF image, and the log entries.
Hopefully, that will give us a lot more information to find the issue you're experiencing.

**Comment by riversdev0** [comment]
@zuka1337 Any update on this? There's a set of improvements coming for Spikekill, but I want to make sure this issue is addressed in the updates.

**Comment by zuka1337** [comment]
@riversdev0 Nothing changed, I was waiting to reproduce the tests above. but still, I can't remove spikes from a single graph with 2 or more data sources associeted (rrds).

Works great on single (one unity data source, rrd)..

I've done an sh script so I can remove the spikes without having to do it manually like removing high values ​​by hand of an xml dump.

Graphs Bellow:
"spike" one has only one data source associated, one OID, one single source of data. Works.
"spike 1" has two data sources, 2 OID, 2 sources. Does not work.

**Ps: I'm realy sorry if I can't make it clear enough for you..**

![spike_working](https://user-images.githubusercontent.com/57998098/89191883-781c7700-d59b-11ea-825f-24607d3cf0ad.gif)
```
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 1.0000000000e+05 with NaN
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 1.0000000000e+05 with NaN
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 1.0000000000e+05 with NaN
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 1.0000000000e+05 with NaN
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 1.0000000000e+05 with NaN
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 1.0000000000e+05 with NaN
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 1.0000000000e+05 with NaN
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 1.0000000000e+05 with NaN
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 1.0233333333e+00 with NaN
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 2.0533333333e+00 with NaN
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 1.0000000000e+05 with NaN
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 1.0000000000e+05 with NaN
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 8.3260701389e+03 with NaN
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 3.3281285556e+04 with NaN
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 1.0000000000e+05 with NaN
2020-08-03 15:09:05 - SPIKEKILL DEBUG: replacing dsvalue 1.0000000000e+05 with NaN
2020-08-03 15:09:05 - SPIKE DEBUG: Limited to window: Aug 3, 2020 14:25:21 thru Aug 3, 2020 14:57:01
2020-08-03 15:09:05 - SPIKE DEBUG: number of DSes: 1
2020-08-03 15:09:05 - SPIKE DEBUG: number of RRAs: 16
2020-08-03 15:09:05 - SPIKEKILL CactiUser:admin, File:cetpcact002_spike_44.rrd, Method:RangeFloat, OutStart:1596461121, OutEnd:1596463021, AvgNan:last

```




![spike_not](https://user-images.githubusercontent.com/57998098/89192451-57a0ec80-d59c-11ea-9442-ff1868c8f1c2.gif)

```
2020-08-03 15:13:08 - WEBLOG CACTI2RRD: /usr/bin/rrdtool info /var/www/html/cacti/rra/cetpcact002_spike_42.rrd
--
2020-08-03 15:13:08 - WEBLOG CACTI2RRD: /usr/bin/rrdtool info /var/www/html/cacti/rra/cetpcact002_spike1_43.rrd
2020-08-03 15:13:08 - SPIKE DEBUG: number of DSes: 1
2020-08-03 15:13:08 - SPIKE DEBUG: number of RRAs: 16
2020-08-03 15:13:07 - SPIKEKILL CactiUser:admin, File:cetpcact002_spike_42.rrd, Method:Variance, AvgNan:last, Kills:100, Outliers:1000, Percent:5000
2020-08-03 15:13:07 - SPIKE DEBUG: number of DSes: 1
2020-08-03 15:13:07 - SPIKE DEBUG: number of RRAs: 10
2020-08-03 15:13:07 - SPIKEKILL CactiUser:admin, File:cetpcact002_spike1_43.rrd, Method:Variance, AvgNan:last, Kills:100, Outliers:1000, Percent:5000
2020-08-03 15:13:05 - WEBLOG CACTI2RRD: /usr/bin/rrdtool graphv - --imgformat=PNG --start='1596462817' --end='1596463585' --pango-markup --title='spike1' --slope-mode --base=1000 --height=150 --width=500 --alt-autoscale-max --lower-limit='0' COMMENT:"From 2020-08-03 14\:53\:37 To 2020-08-03 15\:06\:25\c" COMMENT:" \n" --color BACK#F3F3F3 --color CANVAS#FDFDFD --color SHADEA#CBCBCB --color SHADEB#999999 --color GRID#C4C4C4 --color MGRID#1A1C1C --color FONT#000000 --color AXIS#2C4D43 --color ARROW#2C4D43 --color FRAME#2C4D43 --border 1 --font TITLE:11:'Arial' --font AXIS:8:'Arial' --font LEGEND:8:'Courier' --font UNIT:8:'Arial' --font WATERMARK:6:'Arial' --slope-mode --watermark 'Generated by Cacti®' DEF:a='/var/www/html/cacti/rra/cetpcact002_spike1_43.rrd' DS[spike1] Graphs[spike1]:'spike1':AVERAGE DEF:b='/var/www/html/cacti/rra/cetpcact002_spike_42.rrd' DS[spike] Graphs[spike1]:'spike':AVERAGE AREA:a#0000FFFF:'Outbound\:' AREA:b#C02942FF:'inbound\: '
2020-08-03 15:13:05 - WEBLOG CACTI2RRD: /usr/bin/rrdtool info /var/www/html/cacti/rra/cetpcact002_spike_42.rrd
2020-08-03 15:13:05 - WEBLOG CACTI2RRD: /usr/bin/rrdtool info /var/www/html/cacti/rra/cetpcact002_spike1_43.rrd
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 8.9933333333e+00 with NaN
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 6.9733333333e+00 with NaN
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 8.9933333333e+00 with NaN
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 6.9733333333e+00 with NaN
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 8.9933333333e+00 with NaN
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 6.9733333333e+00 with NaN
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 8.9933333333e+00 with NaN
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 6.9733333333e+00 with NaN
2020-08-03 15:13:05 - SPIKE DEBUG: Limited to window: Aug 3, 2020 14:53:37 thru Aug 3, 2020 15:06:25
2020-08-03 15:13:05 - SPIKE DEBUG: number of DSes: 1
2020-08-03 15:13:05 - SPIKE DEBUG: number of RRAs: 16
2020-08-03 15:13:05 - SPIKEKILL CactiUser:admin, File:cetpcact002_spike_42.rrd, Method:RangeFloat, OutStart:1596462817, OutEnd:1596463585, AvgNan:last
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 4.3903333333e+01 with NaN
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 1.5180000000e+01 with NaN
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 4.3903333333e+01 with NaN
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 1.5180000000e+01 with NaN
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 4.3903333333e+01 with NaN
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 1.5180000000e+01 with NaN
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 4.3903333333e+01 with NaN
2020-08-03 15:13:05 - SPIKEKILL DEBUG: replacing dsvalue 1.5180000000e+01 with NaN
2020-08-03 15:13:05 - SPIKE DEBUG: Limited to window: Aug 3, 2020 14:53:37 thru Aug 3, 2020 15:06:25
2020-08-03 15:13:05 - SPIKE DEBUG: number of DSes: 1
2020-08-03 15:13:05 - SPIKE DEBUG: number of RRAs: 10
2020-08-03 15:13:05 - SPIKEKILL CactiUser:admin, File:cetpcact002_spike1_43.rrd, Method:RangeFloat, OutStart:1596462817, OutEnd:1596463585, AvgNan:last
2020-08-03 15:13:03 - WEBLOG CACTI2RRD: /usr/bin/rrdtool graphv - --imgformat=PNG --start='1596462817' --end='1596463585' --pango-markup --title='spike1' --slope-mode --base=1000 --height=150 --width=500 --alt-autoscale-max --lower-limit='0' COMMENT:"From 2020-08-03 14\:53\:37 To 2020-08-03 15\:06\:25\c" COMMENT:" \n" --color BACK#F3F3F3 --color CANVAS#FDFDFD --color SHADEA#CBCBCB --color SHADEB#999999 --color GRID#C4C4C4 --color MGRID#1A1C1C --color FONT#000000 --color AXIS#2C4D43 --color ARROW#2C4D43 --color FRAME#2C4D43 --border 1 --font TITLE:11:'Arial' --font AXIS:8:'Arial' --font LEGEND:8:'Courier' --font UNIT:8:'Arial' --font WATERMARK:6:'Arial' --slope-mode --watermark 'Generated by Cacti®' DEF:a='/var/www/html/cacti/rra/cetpcact002_spike1_43.rrd' DS[spike1] Graphs[spike1]:'spike1':AVERAGE DEF:b='/var/www/html/cacti/rra/cetpcact002_spike_42.rrd' DS[spike] Graphs[spike1]:'spike':AVERAGE AREA:a#0000FFFF:'Outbound\:' AREA:b#C02942FF:'inbound\: '
2020-08-03 15:13:03 - WEBLOG CACTI2RRD: /usr/bin/rrdtool info /var/www/html/cacti/rra/cetpcact002_spike_42.rrd
2020-08-03 15:13:03 - WEBLOG CACTI2RRD: /usr/bin/rrdtool info /var/www/html/cacti/rra/cetpcact002_spike1_43.rrd
```

**Comment by zuka1337** [comment]
Btw, the tests above, the graphs lose history every time I try this.
So I cant rly aplly on PROD.

**Comment by riversdev0** [comment]
Based upon the debugs, it looks like you're hitting multiple problems that are resolved in the `develop` version of Spikekill.  This version is undergoing testing and will be released in Cacti 1.3, if testing passes.
 
If you'd like to grab an early copy of the `lib/spikekill.php` script from the 'develop' branch, you are welcome to do so, and then save it on top of your existing `lib/spikekill.php` script.  Here's a link: https://raw.githubusercontent.com/Cacti/cacti/develop/lib/spikekill.php

I would recommend that you do that, then run your tests again and see if the results are better.  If they are not better, reply back with updated test results (including debug output).

**Comment by netniV** [comment]
You will need to be careful though, going forward simply taking from 1.3 will not be so straight forward because we are likely to be introducing typing to functions to ensure parameters are what we expect at least to some small degree. 

Also anything reference versions now use a defined variable rather than a function. And there are some other changes that are in there which means you shouldn't simply keep "stealing" from one branch and the other. 

In this case, you may be OK but as there were changes to some of the other files, should as JavaScript and html output, I am dubious over this. I'll defer to river since he knows this bit of code better than me at the moment 😃

**Comment by zuka1337** [comment]
Thank you guys !
I'll make the tests later..
For now i've done a simple frontend/backend to workaround this problem.
It can save time: https://github.com/zuka1337/killspikealt

**Comment by TheWitness** [comment]
Please generate a Pull request if you want this formally fixed.

**Comment by netniV** [comment]
This is fixed I believe in 1.3.  This hasn't been retrofitted due to the extensive change that was done to make it work and needs proper testing.
